lib/rmt/cli/products.rb

```diff
@@ -1,32 +1,37 @@
-# rubocop:disable Rails/Output
-
 require 'rmt/cli/repos'
 
 class RMT::CLI::Products < RMT::CLI::Base
 
-  desc 'list', 'List all products'
+  default_task :list
+
+  desc 'list', 'List products which are marked to be mirrored.'
+  option :all, aliases: '-a', type: :boolean, desc: 'List all products, including ones which are not marked to be mirrored'
   option :release_stage, aliases: '-r', type: :string, desc: 'beta, released'
   def list
     attributes = %i[id name version arch product_string release_stage mirror? last_mirrored_at]
     headings = ['ID', 'Name', 'Version', 'Architecture', 'Product string', 'Release stage', 'Mirror?', 'Last mirrored']
 
-    rows = Product.with_release_stage(options[:release_stage]).map do |product|
+    products = options.all ? Product.all : Product.mirrored
+    rows = products.with_release_stage(options[:release_stage]).map do |product|
       attributes.map { |a| product.public_send(a) }
     end
 
-    if rows.empty?
-      warn 'No products found in the DB. Run "rmt-cli scc sync" to synchronize with your SUSE Customer Center data first.'
+    puts Terminal::Table.new(headings: headings, rows: rows) unless rows.empty?
+
+    if options.all
+      warn 'Run "rmt-cli sync" to synchronize with your SUSE Customer Center data first.' if rows.empty?
     else
-      puts Terminal::Table.new headings: headings, rows: rows
+      warn 'No matching products found in the database.' if rows.empty?
+      puts 'Only enabled products are shown by default. Use the `--all` option to see all products.'
     end
```

Comment:
`rows.empty?` gets called in 2 `if`s and 1`unless`...
How about...

```ruby
    if rows.empty?
      if options.all
        warn 'Run "rmt-cli sync" to synchronize with your SUSE Customer Center data first.'
      else
        warn 'No matching products found in the database.'
      end
    else
      puts Terminal::Table.new(headings: headings, rows: rows)
      puts 'Only enabled products are shown by default. Use the `--all` option to see all products.' unless options.all
    end
```

Reply:
Thanks for the suggestion. I know my version looked overly complicated.
But in yours, you missed the case where it outputs `Only enabled products are shown by default. Use the `--all` option ...` when you you have `rows.empy?` and not used `--all`. So I moved that line out of that if-block ... hope it is okay now.